src/views/Generator/RuleEditor/CorrelationEditor.tsx

```diff
@@ -34,6 +35,29 @@ export function CorrelationEditor() {
         <Text size="2" as="p" mb="2" color="gray">
           Extraction value for correlation.
         </Text>
+        <Box mb="3">
+          <Label>Extraction Mode</Label>
+          <Select.Root
+            defaultValue="multiple"
+            value={extractionMode}
+            onValueChange={(value: 'single' | 'multiple') => 
+              setValue('extractionMode', value)
+            }
+          >
+            <Select.Trigger />
+            <Select.Content>
+              <Select.Item value="single">
+                Single (extract once)
+              </Select.Item>
+              <Select.Item value="multiple">
+                Multiple (extract every time)
+              </Select.Item>
```

Comment:
If Single/Multiple is not descriptive enough, why not call the options `Extract once` and `Extract every time`/`Extract every occurrence`? Maybe even `Extract first occurrence`/`Extract every occurrence` because it gives a better idea of which occurrence/value will be extracted than `once`.